### PR TITLE
refactor(http3): use http3 instead of quic

### PR DESCRIPTION
**Pull request by zll600**
### Description

In https://github.com/apache/apisix/pull/10989, I add `enable_quic` option to `config-default.yaml`.  But `enable_quic` does not match the mainstream gateway configuration. Both Cloudflare and AWS use `enable_http3`. So I want to change `enable_quic` to `enable_http3`.

For enable `HTTP/3` in nginx, we have two steps to do.
1. enable `quic` in a `listen` directive for one port.
2. enable `http3` directive in `server`/`http` directive

eg. 
```nginx
    server {
        http3 on;
        listen 8443 quic reuseport;
        listen 8443 ssl;
```

Here I give two examples

`config-defaul.yaml`
```yaml
apsix:
  ssl:
    enable: true
    listen:                                       # APISIX listening port for HTTPS traffic.
      - port: 9443
        enable_http2: true
        enable_http3: false                       # Enable HTTP/3 (with QUIC). If not set default to `false`.
      # - ip: 127.0.0.3                           # If not set, default to `0.0.0.0`.
      #   port: 9445
      #   enable_http2: true
      #   enable_http3: true
    # ssl_trusted_certificate: /path/to/ca-cert   # Set the path to CA certificates used to verify client
                                                  # certificates in the PEM format.
    ssl_protocols: TLSv1.2 TLSv1.3                # TLS versions supported.
```

#### exmaple 1: enable http3 for two ports
`config.yaml`
```yaml
apisix:
  ssl:
    listen:
      - port: 9443
        enable_http2: true
        enable_http3: true   # access  https://127.0.0.1:9443 with HTTP/3
      - port: 9444
        enable_http2: true
        enable_http3: true   # access https://127.0.0.1:9444 with HTTP/3
```
generated `nginx.conf`
```nginx
    server {
        http2 on;
        http3 on;
        listen 0.0.0.0:9080 default_server reuseport;
        listen [::]:9080 default_server reuseport;
        listen 0.0.0.0:9443 quic default_server reuseport;
        listen 0.0.0.0:9443 ssl default_server;
        listen [::]:9443 quic default_server reuseport;
        listen [::]:9443 ssl default_server;
        listen 0.0.0.0:9444 quic default_server reuseport;
        listen 0.0.0.0:9444 ssl default_server;
        listen [::]:9444 quic default_server reuseport;
        listen [::]:9444 ssl default_server;

        server_name _;

        ssl_certificate      cert/ssl_PLACE_HOLDER.crt;
        ssl_certificate_key  cert/ssl_PLACE_HOLDER.key;
        ssl_session_cache    shared:SSL:20m;
        ssl_session_timeout 10m;

        ssl_protocols TLSv1.2 TLSv1.3;
```

#### exmaple2: enable http3 for one port and disable http3 for another port.
`config.yaml`
```yaml
apisix:
  ssl:
    listen:
      - port: 9443
        enable_http2: true
        enable_http3: true   # access https://127.0.0.1:9443 with HTTP/3
      - port: 9444
        enable_http2: true
        # enable_http3: true
```

generated `nginx.conf`
```nginx
    server {
        http2 on;
        http3 on;
        listen 0.0.0.0:9080 default_server reuseport;
        listen [::]:9080 default_server reuseport;
        listen 0.0.0.0:9443 quic default_server reuseport;
        listen 0.0.0.0:9443 ssl default_server;
        listen [::]:9443 quic default_server reuseport;
        listen [::]:9443 ssl default_server;
        listen 0.0.0.0:9444 ssl default_server reuseport;
        listen [::]:9444 ssl default_server reuseport;

        server_name _;

        ssl_certificate      cert/ssl_PLACE_HOLDER.crt;
        ssl_certificate_key  cert/ssl_PLACE_HOLDER.key;
        ssl_session_cache    shared:SSL:20m;
        ssl_session_timeout 10m;

        ssl_protocols TLSv1.2 TLSv1.3;
```

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [ ] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [ ] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: apisix/cli/ngx_tpl.lua [hunk 1]
```diff
@@ -631,12 +631,15 @@ http {
         {% if enable_http2 then %}
         http2 on;
         {% end %}
+        {% if enable_http3_global then %}
```
Comment (moonming) [inline apisix/cli/ngx_tpl.lua:634]: Why is there a `global` suffix?

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix/cli/ngx_tpl.lua (hunk 1 above)
Comment (monkeyDluffy6017) [reply apisix/cli/ngx_tpl.lua:634]: Does that mean http3 is enabled globally?

Comment (zll600) [reply apisix/cli/ngx_tpl.lua:634]: > Does that mean http3 is enabled globally?

yes, I use `enable_http3` to determine whether to add the `quic` parameter in [listen directive](https://nginx.org/en/docs/http/ngx_http_core_module.html)
> The quic parameter (1.25.0) configures the port to accept [QUIC](https://nginx.org/en/docs/http/ngx_http_v3_module.html) connections.

I use `enable_http3_global` to determine whether to enable [http3 directive](https://nginx.org/en/docs/http/ngx_http_v3_module.html) 
> Syntax:	http3 on | off;
Default:	
http3 on;
Context:	http, server

File: t/APISIX.pm [hunk 2]
```diff
@@ -671,7 +671,6 @@ _EOC_
     $a6_ngx_directives
 
     server {
-        listen 1983 quic reuseport;
```
Comment (shreemaan-abhishek) [inline t/APISIX.pm:674]: why did you remove this?

Comment (zll600) [reply t/APISIX.pm:674]: this is a https upstream server for test. But nginx does not support HTTP/3 with usptream. So this line does not take effect.

File: t/APISIX.pm [hunk 3]
```diff
@@ -730,6 +729,7 @@ _EOC_
         listen 1994 quic reuseport;
         listen 1994 ssl;
         http2 on;
+        http3 on;
```
Comment (shreemaan-abhishek) [inline t/APISIX.pm:732]: Why is this being added now? Why wasn't this added during the main feature PR?

Comment (zll600) [reply t/APISIX.pm:732]: The `http3` directive default is `on`. And I forgot to add it in the main feature PR.

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by moonming — APPROVED** [review]
